src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,7 +26,35 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+        String[] splitFormOfAddress = trimmedAddress.split(",");
+        Block blockAddress ;
+        Street streetAddess ;
+        Unit unitAddress;
+        PostalCode postalCode;
+
+        if(splitFormOfAddress.length == 4 ){
+             blockAddress = new Block(splitFormOfAddress[0]);
+             streetAddess = new Street(splitFormOfAddress[1]);
+             unitAddress = new Unit(splitFormOfAddress[2]);
+             postalCode = new PostalCode(splitFormOfAddress[3]);
+            this.value = blockAddress.getBlockNumber()+"," +streetAddess.getStreetAddress()+"," + unitAddress.getUnitAddress() +"," +postalCode.getPostalCode();
+        }else if (splitFormOfAddress.length ==3){
+             blockAddress = new Block(splitFormOfAddress[0]);
+             streetAddess = new Street(splitFormOfAddress[1]);
+             unitAddress = new Unit(splitFormOfAddress[2]);
+            this.value = blockAddress.getBlockNumber()+"," +streetAddess.getStreetAddress()+"," + unitAddress.getUnitAddress();
+        }else if (splitFormOfAddress.length == 2){
+             blockAddress = new Block(splitFormOfAddress[0]);
+             streetAddess = new Street(splitFormOfAddress[1]);
+            this.value = blockAddress.getBlockNumber()+"," +streetAddess.getStreetAddress();
+        }else if (splitFormOfAddress.length == 1){
+             blockAddress = new Block(splitFormOfAddress[0]);
+            this.value = blockAddress.getBlockNumber();
+        }else{
+
+        }
+
+        //this.value = blockAddress.getBlockNumber()+"," +streetAddess.getStreetAddress()+"," + unitAddress.getUnitAddress() +"," +postalCode.getPostalCode();
```

Comment:
remove unnecessary code completely; you can always recover it from previous versions RCS \#FTW